torch/csrc/distributed/c10d/intra_node_comm.cu

```diff
@@ -139,12 +139,21 @@ static __global__ void oneShotAllReduceKernel(
     size_t N_aligned,
     P2pState** p2pStates,
     at::BFloat16** buffers,
-    size_t rank) {
+    size_t rank,
+    bool fuseInputCopy) {
   const size_t numelPerThread = kBytesPerThread / sizeof(at::BFloat16);
   const size_t offset =
       (blockDim.x * blockIdx.x + threadIdx.x) * numelPerThread;
   const size_t stride = blockDim.x * gridDim.x * numelPerThread;
 
+  if (fuseInputCopy) {
+    for (size_t i = offset; i < N_aligned; i += stride) {
+      bf16x8 val;
+      streamLoad128(val, &input[i]);
+      streamStore128(&buffers[rank][i], val);
+    }
```

Comment:
IIUC, we are loading 8 x bf16 at one time.
But here `stride` seems to be at the unit of bf16. 
Am I missing anything?

Reply:
Hi @kwen2501, the unit of `stride` is `numelPerThread = kBytesPerThread / sizeof(at::BFloat16)` which is also 8 x bf16.